### ISSUE TITLE
Pandora embedded app vs AGDNS

### ISSUE THREAD

**Issue by TPS** [AdguardTeam/AdGuardSDNSFilter#523]
I've recently setup AGDNS-Family in our home modem/router to defeat ads mostly in our Sony blu-ray player apps.  (Our other devices run AG natively with configured encrypted DNS, but, of course, bootstrap DNS from the router, as well.)  However, I've noticed the Pandora app hasn't been even opening since.

Just in case, I attempted to follow the steps @ https://help.pandora.com/s/article/Pandora-on-Sony-Home-Devices-1519949303094 but nothing there helps, & switching router settings to non-blocking (OpenDNS or connection-native) DNS solves it.

What's best to be done?  Should I indeed leave AGDNS off the router, or is there anything that can be tweaked @ your end?

**Comment by TPS** [comment]
Perhaps related, I'm having issues with getting certain YouTube channels' livestreams from playing.  Other livestreams work just fine, &, again, switching DNS solves it.  _(I might change the issue title to reflect this, if y'all wish.)_

**Comment by ameshkov** [comment]
Regarding YT, this might be okay, AG Family is rather strict and it enables "safe mode" for YT, this disables a lot of stuff.

Regarding Pandora, that's a good question. Could you please try installing AdGuard Home on your home PC and using it as a DNS server? This way we could inspect what domain names Pandora connects to and figure out which one is a problem.

**Comment by TPS** [comment]
@ameshkov I will try running AG Home as soon as is practical (will be some time), but am already running AG Windows.  Are they compatible together?

**Comment by ameshkov** [comment]
Well, they are different. AGH is a "server" software, so it controls other devices (that are configured to use it).
AGW is client-side software, it's responsible for one device only.

In this particular case, I suggest trying AGH just as a simple way to inspect DNS queries sent by the Pandora app.

**Comment by TPS** [comment]
I did understand they're different.  But I've only the 1 PC, so I'd like to run both the server & client together on same laptop.

**Comment by ameshkov** [comment]
On just a PC it's enough to run AG alone.

Wait, do you mean that the Pandora app runs on that PC as well? 
In this case DNS queries can be inspected in AdGuard's filtering log

**Comment by TPS** [comment]
No, Pandora runs on the Sony Blu-ray Player.  I only have the 1 laptop on which to run AG-Home, & that PC runs AG-Windows for device-wide blocking.

If I were to _also_ run AG-Home (to serve DNS-blocking to my home network — of which my Blu-ray player is the only device _not_ natively running AG)  _while_ running AG-Windows on that laptop, would that work, or would it be a problem?

**Comment by ameshkov** [comment]
> If I were to also run AG-Home (to serve DNS-blocking to my home network — of which my Blu-ray player is the only device not natively running AG) while running AG-Windows on that laptop, would that work, or would it be a problem?

Nope, they will work just okay.

**Comment by TPS** [comment]
@ameshkov So, I setup AGH with my router & on to my BD player.  There seems to be some sort of deep incompatibility between AGH & the BD player, as the latter refuses to connect online _@ all_ while AGH is running, which is _much_ worse than when trying to run w/ AGDNS Family.  (All the other network devices work just fine when pointing DNS toward AGH.) 
 The only thing that's in the Query Log is
```
16:27:37
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
23 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:30
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
24 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:29
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
24 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:23
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
22 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:22
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
26 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:16
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
23 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:15
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
24 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:11
1/3/2021
doh.opendns.com
Type: AAAA, Plain DNS
Processed
0.00 ms
172.30.0.0
16:27:11
1/3/2021
doh.opendns.com
Type: A, Plain DNS
Processed
0.00 ms
172.30.0.0
16:27:09
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
36 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:08
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
23 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:08
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
22 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:07
1/3/2021
doh.opendns.com
Type: AAAA, Plain DNS
Processed
0.00 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:07
1/3/2021
doh.opendns.com
Type: A, Plain DNS
Processed
0.00 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:04
1/3/2021
doh.opendns.com
Type: AAAA, Plain DNS
Processed
0.00 ms
172.30.0.0
16:27:04
1/3/2021
doh.opendns.com
Type: A, Plain DNS
Processed
0.00 ms
172.30.0.0
16:27:01
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
23 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:01
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
25 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:00
1/3/2021
doh.opendns.com
Type: AAAA, Plain DNS
Processed
27 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:27:00
1/3/2021
doh.opendns.com
Type: A, Plain DNS
Processed
0.00 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:26:54
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
24 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:26:50
1/3/2021
doh.opendns.com
Type: AAAA, Plain DNS
Processed
0.00 ms
172.30.0.0
16:26:50
1/3/2021
doh.opendns.com
Type: A, Plain DNS
Processed
1 ms
172.30.0.0
16:26:47
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
28 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:26:47
1/3/2021
doh.opendns.com
Type: AAAA, Plain DNS
Processed
21 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:26:47
1/3/2021
doh.opendns.com
Type: A, Plain DNS
Processed
21 ms
2601:40a:8003:ce10:68db:12db:ce47:0
16:26:40
1/3/2021
ipv4only.arpa
Type: AAAA, Plain DNS
Processed
100 ms
2601:40a:8003:ce10:68db:
```
So, IDK what's going on or even how to find out. 🤷🏾‍♂️

**Comment by ameshkov** [comment]
Weird.

There're many `ipv4only.arpa` queries, I wonder where do they come from.

Could you please try collecting AdGuard Home debug logs? It will contain more details on DNS queries (including responses).

1. [Configure](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#configuration-file) AdGuard Home to collect logs:
    * Specify `log_file`
    * Set `verbose` to `True`
2. Restart AdGuard Home and reproduce the issue
3. Post the log file here.

**Comment by TPS** [comment]
It'll be 1 of my projects for the coming weekend, I think.  _(The difficulty isn't finding the time in_ my _schedule, but my missus', so I don't disrupt her listening.)_
